src/executor/stack/executor.rs

```diff
@@ -801,33 +792,28 @@ impl<'config, 'precompiles, S: StackState<'config>, P: PrecompileSet>
 		}
 	}
 
-	pub fn initialize_with_access_list(&mut self, access_list: Vec<(H160, Vec<H256>)>) {
+	pub fn initialize_with_access_list(&mut self, access_list: &[(H160, Vec<H256>)]) {
 		let addresses = access_list.iter().map(|a| a.0);
 		self.state.metadata_mut().access_addresses(addresses);
 
 		let storage_keys = access_list
-			.into_iter()
-			.flat_map(|(address, keys)| keys.into_iter().map(move |key| (address, key)));
+			.iter()
+			.flat_map(|(address, keys)| keys.iter().map(move |key| (*address, *key)));
```

Comment:
I don't know that switching to slices is a positive change because now there is a dereference (copy) where there was not previously. I think it's probably better to leave `access_list: Vec` everywhere because then there is no breaking change to the API and it's unclear to me that passing a slice is a better API anyway since the code did make use of the owned collection.

Reply:
Take a note, that H160, H256 wrapping for `&[u8; N]`, and dereference costs nothing.
But I passed `aurora-engine` tests, and there is no profit for NEAR gas costs. Breaking changes API at that point of view is a good argument.

Reply:
Changed: 94cf73ff865e35ed584882775795a7e74462e638